src/app/window-handler.ts

```diff
@@ -259,27 +247,38 @@ export class WindowHandler {
             // Injects custom title bar css into the webContents
             // only for Window and if it is enabled
             await injectStyles(this.mainWindow, this.isCustomTitleBarAndWindowOS);
-            if (this.isCustomTitleBarAndWindowOS) {
-                this.mainWindow.webContents.send('initiate-custom-title-bar');
-            }
 
             this.mainWindow.webContents.send('page-load', {
                 isWindowsOS,
                 locale: i18n.getLocale(),
                 resources: i18n.loadedResources,
                 origin: this.globalConfig.url,
+                enableCustomTitleBar: this.isCustomTitleBarAndWindowOS,
             });
             this.appMenu = new AppMenu();
+        });
 
-            // close the loading window when
-            // the main windows finished loading
-            if (this.loadingWindow) {
-                this.loadingWindow.destroy();
-                this.loadingWindow = null;
-            }
+        this.mainWindow.webContents.on('did-fail-load', (_event, errorCode, errorDesc, validatedURL) => {
+            logger.error(`Failed to load ${validatedURL}, with an error: ${errorCode}::${errorDesc}`);
+            this.loadFailError = errorDesc;
+        });
 
-            // Ready to show the window
-            this.mainWindow.show();
+        this.mainWindow.webContents.on('did-stop-loading', () => {
+            if (this.mainWindow && windowExists(this.mainWindow)) {
+            this.mainWindow.webContents.executeJavaScript('document.location.href').then((href) => {
```

Comment:
will this have any security issues?

Reply:
I don't think so. This block of code is executed by `electron` and the parameter is hardcoded